### PR TITLE
Switch chart engine from nvd3 to c3

### LINKED ISSUE #3571: Use c3 instead of nvd3
[nvd3](https://github.com/novus/nvd3) is not maintained for a long time and uses a very old version of [d3](https://github.com/d3/d3). We should switch to something that is maintained and frequently updated, i.e. [c3](https://github.com/c3js/c3).

### PR DESCRIPTION
**Pull request by eloyekunle**
Node Detail:
image

Workload Status:
image

Closes #3571

### CODE REVIEW

**Comment by maciaszczykm** [comment]
Hey, any updates?

**Comment by eloyekunle** [comment]
Hi @maciaszczykm, No updates yet, but by next week I'll be implementing the POC and the feedback from https://github.com/kubernetes/dashboard/issues/3571#issuecomment-473293299 into this PR.

So as discussed in #3571, this would be a switch from `nvd3` to `d3`.

**Comment by eloyekunle** [comment]
@maciaszczykm The PR is ready for review.

File: src/app/frontend/common/components/allocationchart/component.ts [hunk 1]
```diff
@@ -12,158 +12,150 @@
...
   @Input() enableTooltips = false;
-  @Input() size: number;
+  @Input() size = 280;
   @Input() id: string;
 
-  ngOnInit(): void {
-    this.generateGraph_();
+  allocated = new Set();
+
+  ngAfterViewInit(): void {
```
Comment (ajatprabha) [inline src/app/frontend/common/components/allocationchart/component.ts]: Just curious, why can't we use `ngOnInit()` here?

**Comment by maciaszczykm** [comment]
@eloyekunle Looks promising! Can you solve the conflicts?

**Comment by floreks** [comment]
@eloyekunle PR lgtm. Rebase please and we can merge.

File: src/app/frontend/common/components/allocationchart/component.ts (hunk 1 above)
Comment (eloyekunle) [reply src/app/frontend/common/components/allocationchart/component.ts]: @ajatprabha It wasn't displaying the chart with `ngOnInit()`.

**Comment by maciaszczykm** [comment]
You have to commit `package-lock.json` as you have also done some changes to the `package.json`. CI is failing due to that.

**Comment by eloyekunle** [comment]
@maciaszczykm Fixed now.

**Pull request merged**